src/nisarqa/processing/insar/quiver_plots.py

```diff
@@ -382,4 +568,282 @@ def truncate_colormap(
+                wavelength=wavelength,
+                side=look_side,
+                **geo2rdr_params,
+            )
+            target_in_rdr_matrix[i, j, 0] = aztime
+            target_in_rdr_matrix[i, j, 1] = srange
+
+    # 3) Use the az/rng offset values (in meters) at the coordinate
+    #    to offset ("shift") the point in the radar grid
+
+    # Azimuth Offsets are in meters, but azimuth on the radar grid is in
+    # seconds. So, convert `az_offsets_at_arrow_tails` from meters to seconds
+    # by using the ground track velocity (meters/second) metadata cube.
+    X_tail_vals_grid, Y_tail_vals_grid = np.meshgrid(
+        x_coords_values_at_arrow_tails, y_coords_values_at_arrow_tails
+    )
+
+    # Constructing the interpolator is an expensive operation (1-2 seconds),
+    # so do this outside of the for loops.
+    grd_trk_vel_at_arrow_tail = nisarqa.interpolate_points_in_metadata_cube(
+        data=ground_track_velocity.data,
+        height_coordinates=ground_track_velocity.z_coord_vector,
+        y_coordinates=ground_track_velocity.y_coord_vector,
+        x_coordinates=ground_track_velocity.x_coord_vector,
+        # assume zero-height again
+        h_vals_of_points=np.zeros(
+            (num_arrows_y_direction, num_arrows_x_direction)
+        ),
+        y_vals_of_points=Y_tail_vals_grid,
+        x_vals_of_points=X_tail_vals_grid,
+        method="cubic",  # GOFF Product Specs recommend "cubic"
+    )
+
+    az_off_values_at_arrow_tail_in_seconds = (
+        az_offsets_at_arrow_tails / grd_trk_vel_at_arrow_tail
+    )
+
+    # Shift the radar grid's azimuth values
+    target_in_rdr_matrix[:, :, 0] += az_off_values_at_arrow_tail_in_seconds
+
+    # Shift the radar grid's range values
+    # Note: range offset values were already in meters, so just add them
+    target_in_rdr_matrix[:, :, 1] += rg_offsets_at_arrow_tails
+
+    # 4) Run geo2rdr to convert back to (x_1, y_1) LLH
+    x_tail_coords_shifted = np.zeros(
+        (num_arrows_y_direction, num_arrows_x_direction),
+        dtype=y_coords_values_at_arrow_tails.dtype,
+    )
+    y_tail_coords_shifted = np.zeros(
+        (num_arrows_y_direction, num_arrows_x_direction),
+        dtype=y_coords_values_at_arrow_tails.dtype,
+    )
+    for i in range(num_arrows_y_direction):
+        for j in range(num_arrows_x_direction):
+            aztime = target_in_rdr_matrix[i, j, 0]
+            srange = target_in_rdr_matrix[i, j, 1]
+            # Skip NaN pixels (this is usually geocoding fill)
+            if not np.isfinite(aztime) or not np.isfinite(srange):
+                x_tail_coords_shifted[i, j] = np.nan
+                y_tail_coords_shifted[i, j] = np.nan
+                continue
+
+            xyz = isce3.geometry.rdr2geo_bracket(
+                # az time in seconds since orbit.reference_epoch
+                aztime=target_in_rdr_matrix[i, j, 0],
+                # slant range in meters
+                slant_range=target_in_rdr_matrix[i, j, 1],
```

Comment:
```suggestion
                aztime=aztime,
                # slant range in meters
                slant_range=srange,
```

You already assigned the variables `aztime` and `srange` above, so you might as well use those variables here.

https://github.com/isce-framework/nisarqa/blob/219b355eb9011e69a3fe0eb359fa1e8fdfaabba6/src/nisarqa/processing/insar/quiver_plots.py#L798-L799

Reply:
d'oh! Good catch. Thanks!